### PR TITLE
Add custom event handler docs

### PR DESCRIPTION
**Pull request by zhaomin1423**
### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by zhaomin1423** [comment]
The custom handler will be added after https://github.com/apache/incubator-kyuubi/pull/3090 is merged.

File: docs/extensions/server/events.rst [hunk 1]
```diff
@@ -18,5 +18,47 @@ Handle Events with Custom Event Handler
...
-   This page is still in-progress.
+Custom Event Handler
+--------------------
+
+Kyuubi supports custom event handler. It is usually used to write Kyuubi events to some external systems. For example, Kafka, ElasticSearch, etc.
+
+The steps of injecting custom event handler
+--------------------------------------
+
+1. create a custom class which implements the ``org.apache.kyuubi.events.handler.CustomEventHandlerProvider``, and add a file named org.apache.kyuubi.events.handler.CustomEventHandlerProvider in the src/main/resources/META-INF/services folder of project, its content is the custom class name.
```
Comment (yaooqinn) [inline docs/extensions/server/events.rst:29]: don't we need to reference libraries first?

PS, use non-ordered list.

**Comment by yaooqinn** [comment]
A good example to follow https://kyuubi.readthedocs.io/en/latest/extensions/server/authentication.html

1. reference the built-in implementations for a better understanding of what the plugin is and whether a user really needs a custom one.
2. All developer APIs are elaborated with rich java doc


A bad example to avoid (you just followed it)
https://kyuubi.readthedocs.io/en/latest/extensions/server/configuration.html

The doc hierarchy goes

```
H1 What is it?
  
  *introduction about this feature and related APIS*
 
  H2 the way to develop
     *develop guide*
  H2 the way to activate
     *usage guide*
```

**Comment by zhaomin1423** [comment]
> A good example to follow https://kyuubi.readthedocs.io/en/latest/extensions/server/authentication.html
> 
> 1. reference the built-in implementations for a better understanding of what the plugin is and whether a user really needs a custom one.
> 2. All developer APIs are elaborated with rich java doc
> 
> A bad example to avoid (you just followed it) https://kyuubi.readthedocs.io/en/latest/extensions/server/configuration.html
> 
> The doc hierarchy goes
> 
> ```
> H1 What is it?
>   
>   *introduction about this feature and related APIS*
>  
>   H2 the way to develop
>      *develop guide*
>   H2 the way to activate
>      *usage guide*
> ```
 
Good guide. I updated the doc, also add a sample code to https://github.com/kyuubilab?

**Comment by ulysses-you** [comment]
@zhaomin1423 please update the docs according to the behavior of merged pr. For example, we will ignore the custom event handler if it failed to initialize.

**Comment by ulysses-you** [comment]
thanks, merging to master

**Pull request closed without merging**